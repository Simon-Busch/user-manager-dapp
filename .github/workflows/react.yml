# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions

name: React CI/CD

on: [pull_request, push] 
  # pull_request:
  #   branches: [ main ]

jobs:
  build:

    runs-on: ubuntu-latest

    strategy:
      matrix:
        node-version: [16.x]
        # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
    env:
      URL_INFURA: ${{ secrets.URL_INFURA }}
      ACCOUNT_PRIVATE: ${{ secrets.ACCOUNT_PRIVATE }}

    steps:
    - uses: actions/checkout@v2
    - name: Use Node.js ${{ matrix.node-version }}
      uses: actions/setup-node@v2
      with:
        node-version: ${{ matrix.node-version }}
        cache: 'npm'
    - name: Install all dependencies
      run: npm i 
    - name: Launch the test on React app
      run: npm test
    - name: Launch hardhat tests on smart contract
      run: npx hardhat test
