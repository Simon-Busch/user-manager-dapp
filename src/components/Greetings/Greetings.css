.greetings-container {
  padding: 1rem;
}