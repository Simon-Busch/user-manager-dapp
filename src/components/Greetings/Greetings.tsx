import React from "react";
import "./Greetings.css";
interface GreetingsProps {
  account: string | null;
}

const Greetings: React.FC<GreetingsProps> = ({ account }) => {
  return (
    <div className="greetings-container">
      <p className="paragraph-grey">Hey ! Welcome back {account} ✌🏼</p>
    </div>
  );
};

export default Greetings;
