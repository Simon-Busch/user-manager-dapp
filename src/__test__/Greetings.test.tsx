import { render, screen } from '@testing-library/react';
import Greetings from '../components/Greetings/Greetings';

test('renders greetings component', () => {
  render(<Greetings />);
  const welcomeElement = screen.getByText(/Hey ! Welcome back/i);
  expect(welcomeElement).toBeInTheDocument();
});

describe("when rendered with an account prop", () => {
  it("should paste it into the greetings text", () => {
    const { container } = render(<Greetings account="john" />); 
    expect(container.firstChild).toMatchSnapshot();
  });
});