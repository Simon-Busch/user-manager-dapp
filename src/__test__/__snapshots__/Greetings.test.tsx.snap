// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when rendered with an account prop should paste it into the greetings text 1`] = `
<div
  class="greetings-container"
>
  <p
    class="paragraph-grey"
  >
    Hey ! Welcome back 
    john
     ✌🏼
  </p>
</div>
`;
